pyracf/resource/resource_admin.py

```diff
@@ -285,6 +286,133 @@ def get_user_access(
         self.set_running_userid(original_userid)
         return self._get_field(profile, "base", "yourAccess")
 
+    # ============================================================================
+    # Auditing Rules
+    # ============================================================================
+    def get_audit_rules(
+        self, resource: str, class_name: str
+    ) -> Union[dict, bytes, None]:
+        """Get the auditing rules associated with this general resource profile."""
+        profile = self.extract(resource, class_name, profile_only=True)
+        return self._get_field(profile, "base", "auditing")
+
+    def overwrite_audit_by_attempt(
+        self,
+        resource: str,
+        class_name: str,
+        success: Union[str, None] = None,
+        failure: Union[str, None] = None,
+        all: Union[str, None] = None,
+    ) -> Union[dict, bytes]:
+        """
+        Overwrites the auditing rules for this general resource profile with new
+        rules to audit based on specified access attempts.
+        """
+        self.__validate_access_levels(success, failure, all)
+        traits = {}
+        if success is not None:
+            traits[f"base:audit_{success}"] = "success"
+        if failure is not None:
+            traits[f"base:audit_{failure}"] = "failure"
+        if all is not None:
+            traits[f"base:audit_{all}"] = "all"
+        result = self.alter(resource, class_name, traits=traits)
+        return self._to_steps(result)
+
+    def alter_audit_by_attempt(
```

Comment:
Does it make sense to update the naming convention on these to `audit_rules_by` instead of `audit_by`?

Reply:
This did make some test variables VERY long, but tests seem to still pass after what i've done to keep the lines short.

Reply:
actually, black formatter undid my changes, so I'm not sure there's an appropriate way to keep some of these test constants within an appropriate length. I would have to either amend the naming scheme or make special consideration to the following test constants:

```
TestResourceConstants.TEST_RESOURCE_OVERWRITE_AUDIT_RULES_BY_ACCESS_LEVEL_NONE_REQUEST_XML,
TestResourceConstants.TEST_RESOURCE_OVERWRITE_AUDIT_RULES_BY_ACCESS_LEVEL_MULT_REQUEST_XML,
TestResourceConstants.TEST_RESOURCE_OVERWRITE_AUDIT_RULES_BY_ACCESS_LEVEL_ALL_REQUEST_XML,
```

I am alright making changes like shortening OVERWRITE to OVW for these constants and adding a comment in our constant definitions that would explain why.

Reply:
Or i can leave in the exceptions.

Reply:
I think we definitely want to resolve `E501`. I will mostly leave it up to how to shorten the variables. Unfortunately, we will need to change at least one word in the variable name to an acronym.

Reply:
Maybe change `OVERWRITE` to `OVWT`?

Reply:
Actually maybe you can even just remove the word `RULES`. I think that would be acceptable for a test variable.

Reply:
This would be a rare exception to trying to make test constant names and test case name match the functions they are testing.